### PR TITLE
Pull Request for issues #18, #19, #7, #8

### PR DESCRIPTION
**Pull request by MetLob**
Also fixed "Tile Index" maximum value in OAM Editor...
This is commit to close Issue #18 and #19 

For all oldest plugins, where using SpriteBase and OAM Editor, nothing affected and their works has't  been changed.

### CODE REVIEW

**Comment by MetLob** [comment]
Added implementation for Issues #7 and #8

**Comment by pleonex** [comment]
Thanks a lot! and sorry for forgetting about this. I will review all the changes today

File: Plugins/Images/Images/NCER.cs [hunk 1]
```diff
@@ -357,6 +384,13 @@ void Update_Struct()
             ncer.cebk.section_size = 0x20 + size;
             if (ncer.cebk.section_size % 4 != 0)
                 ncer.cebk.section_size += (4 - (ncer.cebk.section_size % 4));
+            
+            // Update partition data info
+            ncer.cebk.partition_data_offset = ncer.cebk.section_size - 8;
+            ncer.cebk.max_partition_size = max_partition_size;
+            ncer.cebk.first_partition_data_offset = (ncer.cebk.first_partition_data_offset > 0) ? ncer.cebk.first_partition_data_offset : 8;
```
Comment (pleonex) [inline Plugins/Images/Images/NCER.cs]: Does this means that the `first_partition_data_offset` can't be smaller than 8? In that case, should be check if it's smaller than 8 instad of 0?

**Comment by pleonex** [comment]
Everything look good for me. I am waiting for a reply to my comment before merging.
Thank you so much! This will make a new release. I am adding you to the credits.

**Comment by MetLob** [comment]
In this code I always added one part of data as default.
And if partition data exist then 'ncer.cebk.partition_data_offset = ncer.cebk.section_size - 8;'
this > 0 because ncer.cebk.section_size > 8
-8 because:

br.BaseStream.Position = ncer.header.header_size + ncer.cebk.partition_data_offset + 8; // 8 is a CEBK general header size (magic and size))

If you do not want to add partition data if original file does not contain this, you can change this code to

           // Update partition data info
            if (ncer.cebk.partition_data_offset != 0)
            {
                ncer.cebk.partition_data_offset = ncer.cebk.section_size - 8;
                ncer.cebk.max_partition_size = max_partition_size;
                ncer.cebk.first_partition_data_offset = ncer.cebk.first_partition_data_offset;
                ncer.cebk.section_size += (uint)(8 * (Banks.Length + 1));
            }

**Comment by MetLob** [comment]
about `ncer.cebk.first_partition_data_offset`
This value has min value = 8, because for all partitions data section fisrt 8 bytes is `first_partition_data_offset` and `max_partition_size` and value of `first_partition_data_offset` relative to this section.

value of `first_partition_data_offset` is offset to position where start reading to partitions data
value of `first_partition_data_offset` = 0 if `partition_data_offset == 0` in source file and i add one part.

I'm sorry but we should next fix:

            #region Read partitions data

            if (ncer.cebk.partition_data_offset != 0)
            {
                br.BaseStream.Position = ncer.header.header_size + ncer.cebk.partition_data_offset + 8; // 8 is a CEBK general header size (magic and size)
                ncer.cebk.max_partition_size = br.ReadUInt32();
                ncer.cebk.first_partition_data_offset = br.ReadUInt32();
                br.BaseStream.Position += ncer.cebk.first_partition_data_offset - 8;
                for (int i = 0; i < ncer.cebk.nBanks; i++)
                {
                    ncer.cebk.banks[i].partition_offset = br.ReadUInt32();
                    ncer.cebk.banks[i].partition_size = br.ReadUInt32();
                }
            }

            #endregion

**Comment by MetLob** [comment]
I fixed this

**Comment by pleonex** [comment]
Perfect! Thanks! I am merging and closing the issues.
Regarding the credits (in the About window) do you want to appear as *MetLob* or with another nick?

**Comment by MetLob** [comment]
Yea, MetLob is OK

**Pull request merged**